shapely/geos.py

```diff
@@ -129,7 +129,9 @@ def load_dll(libname, fallbacks=None, mode=DEFAULT_MODE):
         try:
             egg_dlls = os.path.abspath(
                 os.path.join(os.path.dirname(__file__), 'DLLs'))
-            if hasattr(sys, "frozen"):
+            if hasattr(sys, '_MEIPASS'):
+                wininst_dlls = sys._MEIPASS
+            elif hasattr(sys, "frozen"):
```

Comment:
There are two other tests for `frozen` in this module. Do we need the same for those?

Reply:
@sgillies The simplest test application would be a `test.py` with:

```
from shapely import geos
```

Then using pyinstaller (currently need the hook in github master iirc) run `pyinstaller test.py`. This should create `dist/test` directory with a `test` binary inside. Running this should run to import shapely and then return 0.

As for the other frozen tests, in my experience the others are fine the way they are. The OSX one already uses MEIPASS as a backup. The linux one seems to do the equivalent of MEIPASS, since in my experience MEIPASS is set to the directory of the interpreter/executable which is also the directory where all bundled libraries are placed.

Reply:
@sgillies Were you thinking this would be added to the automated travis testing? Do you want me to try to add that? It seems a little odd to do it in shapely's repository, but it also seems unreasonable for pyinstaller to test a ton of the possible packages that someone could try to bundle.

Reply:
@davidh-ssec I would love to have such a test for Travis. To be honest, I'm concerned more about the untested code that exists in Shapely now (having taken on some freezing responsibilities) than I am about the burden on the pyinstaller project, but I'm happy to pick up some of that burden.

Reply:
@davidh-ssec Hi there, I don't have the time to learn about pyinstaller and write the tests, and don't want to commit an untested feature. Are you interested in doing it? Have you found another solution?

Reply:
I will try to find the time to do it. There is a way to test this I think, but like I said it'll seem odd being in the shapely testing. I guess there isn't another way.

I haven't found a different way of doing this I just had built my application using my own forks and then stepped away from that project for a while. I'll try to find some time to do this this week.

Reply:
@sgillies I've been faced with the same issue as @davidh-ssec and arrived at the same solution that he did for building a shapely dependent exe on windows with PyInstaller. In order for you to be comfortable merging these 3 lines, would we have to build this test in AppVeyor in order for `sys.platform == 'win32'` to ever be `True` in the test case? Would we also need to build the test exe for Darwin systems too since those lines don't have a test?

It sounds like checking for and testing all use/build cases is a lot of work that could be avoided by letting the user set/overwrite the path to the dependencies they want `shapely` find. This solution would also be more resilient to changes that are outside Shapely's control, like conventions chosen by conda and PyInstaller that may change with each point release.
pseudocode:
```
import sys
import shapely
shapely.geos.dlls(sys._MEIPASS)
```
I'm a newcomer to this conversation so I don't know if this is feasible (but I am aware that this would need to be its own separate issue/PR and would require a good deal more time to implement). If this type of pattern is possible, it could be fully tested on Travis and eliminate the complicated maintenance required for supporting corner cases like custom installs and bundled executables built by 3rd party compilers. In the meantime, or if this pattern is not possible, maybe we can merge the _MEIPASS lines for windows as a 'non-supported or tested corner case courtesy' rather than treating it as a 'new feature' since windows-only lines are hard to test in Travis, and loading from `sys._MEIPASS` is already provided (though not tested) for Darwin users.

Reply:
@austinorr I think your suggestion is the ultimately the right way to do this and that a better-designed shapely module would invoke such a method (with defaults) right before the first use of the GEOS DLL.

That's more work than I'm able to do or oversee at the moment, and so I'm inclined to merge this PR if @snorfalorpagus agrees and put off any rewrite of shapely.geos to a future version.